src/Storage/Controllers/DataController.cs

```diff
@@ -411,6 +415,29 @@ public async Task<ActionResult<DataElement>> OverwriteData(int instanceOwnerPart
             return Ok(updatedDataElement);
         }
 
+        /// <summary>
+        /// Sets the file scan status for an existing data element.
+        /// </summary>
+        /// <param name="instanceGuid">The id of the instance that the data element is associated with.</param>
+        /// <param name="dataGuid">The id of the data element to update.</param>
+        /// <param name="fileScanStatus">The file scan results for this data element.</param>
+        /// <returns>The updated data element.</returns>
+        [Authorize(Policy = "PlatformAccess")]
+        [HttpPut("dataelements/{dataGuid}/filescanstatus")]
+        [Consumes("application/json")]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status400BadRequest)]
+        [Produces("application/json")]
+        public async Task<ActionResult<DataElement>> SetFileScanStatus(
+            Guid instanceGuid,
+            Guid dataGuid,
+            [FromBody] FileScanStatus fileScanStatus)
+        {
+            var success = await _dataRepository.SetFileScanStatus(instanceGuid.ToString(), dataGuid.ToString(), fileScanStatus);
+
+            return success ? Ok() : BadRequest("An error occurred while updating file scan status.");
```

Comment:
BadRequest should be used when there is something identifiably wrong with the request. The message indicate an unknown error.

Reply:
Will use error 500 for now until we know how cosmos will handle predicate filter mismatch.